HPCC-33405: Add support for ubuntu arm

GordonSmith opened the pull request:
Move antlr3c into vcpkg
DC: Fix for arm64 register naming
Rework dockerfiles/vcpkg/build.sh to support arm

Signed-off-by Gordon Smith <GordonJSmith@gmail.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

GordonSmith commented:
@jackdelv Can you look at the mongodb changes?

dcamper commented on cmake_modules/vcpkg.cmake:
```diff
@@ -19,8 +19,15 @@ elseif(APPLE)
     set(VCPKG_TARGET_TRIPLET "x64-osx" CACHE STRING "target triplet")
   endif()
 elseif(UNIX)
+  execute_process(COMMAND uname -m OUTPUT_VARIABLE ARCHITECTURE)
```
Aside: If we ever want to do cross-compiling, this won't be correct.

GordonSmith replied:
Agree - I had a mental note to see if there was a better way than this (like the OSX flag).

dcamper reviewed (approved):
Looks good. One random comment.

jackdelv reviewed (approved):
@GordonSmith MongoDB changes look good.

ghalliday reviewed (commented):
@GordonSmith the unit test failure looks significant.

ghalliday reviewed (changes requested):
See comment above

dcamper commented:
It looks like there was some merging issues. Parquet tests were failing due primarily to duplicated lines in the ECL code.

testing/regress/ecl/parquetTypes.ecl:65:

`integerDatasetIn := ParquetIO.Read(integerRecord, dropzoneDirectory + 'IntegerTest.parquet');`
`integerDatasetIn := ParquetIO.Read(integerRecord, dropzoneDirectory + '/IntegerTest.parquet');`

GordonSmith committed:
HPCC-33405: Add support for ubuntu arm

Bump vcpkg to 2025.02.13 with arm support
Move antlr3c into vcpkg
DC: Fix for arm64 register naming
DC: Fix arm64 build issue in _cpyrevn
DC: Move large zero array outside of functions (jlib file unittest)
Rework dockerfiles/vcpkg/build.sh to support arm

Signed-off-by Gordon Smith <GordonJSmith@gmail.com>

Move large zero array outside of functions

GordonSmith commented:
@ghalliday rebased and should be ready to merge once the tests have completed.
